### PR TITLE
fix(*): allow_with_shadow_deny rule

### PR DESCRIPTION
**Pull request by lobkovilya**
Signed-off-by: Ilya Lobkov <ilya.lobkov@konghq.com>

### Checklist prior to review

- [X] Link to docs PR or issue --
- [X] Link to UI issue or PR --
- [X] Is the [issue worked on linked][1]? -- Fix https://github.com/kumahq/kuma/issues/5065
- [X] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [X] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Unit Tests --
- [ ] E2E Tests --
- [ ] Manual Universal Tests --
- [X] Manual Kubernetes Tests --
- [X] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? -- No
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? -- No
- [X] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests? No

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/policies/meshtrafficpermission/xds/rbac_configurer_test.go [hunk 1]
```diff
@@ -252,5 +252,35 @@ filters:
...
+      rules:
+          policies:
+              MeshTrafficPermission:
+                  permissions:
+                      - any: true
+                  principals:
+                      - authenticated:
+                          principalName:
+                              exact: spiffe://shadow_deny_mesh/backend
+      shadowRules: {}
```
Comment (jakubdyszkiewicz) [inline pkg/plugins/policies/meshtrafficpermission/xds/rbac_configurer_test.go:282]: I'm lost here. What does empty `shadowRules` mean?

I'd expect that we will shadow deny only `kuma.io/service: backend`

Comment (lobkovilya) [reply pkg/plugins/policies/meshtrafficpermission/xds/rbac_configurer_test.go:282]: Right, I had in mind that the default action for shadow rules is `DENY` as we do with normal rules. But that was wrong assumption, I'll fix it

**Review by jakubdyszkiewicz — APPROVED** [review]
Code is good. However, I miss a trace of the decision why we removed `DENY_WITH_SHADOW_ALLOW`. Should we update MADR?

**Comment by lobkovilya** [comment]
Yes, I'm planing to update it before merging this one

**Comment by lobkovilya** [comment]
xref https://github.com/kumahq/kuma/pull/5129

**Pull request merged**